app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,63 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params)
+    @album.update(:rank => 0)
+    if @album.save
+      redirect_to album_path(@album.id)
+    else
+      render "new"
+    end
+
+  end
+
+  def show
+    @album = Album.find(params[:id])
```

Comment:
Since you repeat `@album = Album.find(params[:id]) on several actions, you can use a `before_action` to set this and DRY up your code.